corc-cascading/src/main/java/com/hotels/corc/cascading/SearchArgumentFactory.java

```diff
@@ -197,6 +203,24 @@ static void checkValueTypes(Fields fields, Object... values) {
       }
     }
 
+    static PredicateLeaf.Type toType(Fields fields) {
```

Comment:
Let's move this method out to a new class - `FieldsTypeUtils` and then write a corresponding unit test for it which tests every branch here.